### PR TITLE
add C++ examples for ctypes

### PR DESCRIPTION
**Pull request by adam-dziedzic**
**Where to put new files:**

- New files should go into a top-level subfolder, named after the article slug. For example: `my-awesome-article`

**How to merge your changes:** 

1. [Make sure the CI code style tests all pass (+ run the automatic code formatter if necessary).](https://github.com/realpython/materials/blob/master/README.md)
2. Find an RP Team member on Slack and ask them to review & approve your PR.
3. Once the PR has one positive ("approved") review, GitHub lets you merge the PR.
4. 🎉

### CODE REVIEW

**Comment by Znunu** [comment]
Hi!
Thanks a lot. While I can't test them, I think they look great. I see you even bothered to make it cross platform which is nice.
There seems to be a slight overlap with [thir](https://github.com/realpython/materials/pull/193) PR. Can you take a look?
I think you need to find jima80525, if you want this merged. He's the one who wrote the article and code files.

File: python-bindings/ctypes_c_test.py [hunk 1]
```diff
@@ -2,13 +2,17 @@
 """ Simple examples of calling C functions through ctypes module. """
 import ctypes
 import sys
+import pathlib
 
 if __name__ == "__main__":
+    libname = pathlib.Path().absolute()
+    print("libname: ", libname)
```
Comment (jima80525) [inline python-bindings/ctypes_c_test.py:9]: This is a good addition! nice

**Review by jima80525 — APPROVED** [review]
Looks good!

**Comment by dbader** [comment]
Cool, LGTM :) @jima80525 Do we need to make any updates to the article before I merge this or is this just a bonus added to the `materials` example code?

**Comment by jima80525** [comment]
@dbader - it changes the output slightly (by adding a `library: libcmult.so` line to the display, but I wouldn't say that's worth an article update.  If it's a big deal, I'd remove it from here, but I don't feel like that will confuse anyone.

**Pull request merged**